cookbook_download.go

```diff
@@ -0,0 +1,108 @@
+//
+//  Author:: Salim Afiune <afiune@chef.io>
+//
+
+package chef
+
+import (
+	"io"
+	"os"
+	"path"
+)
+
+// Download downloads a cookbook to the current directory on disk
+func (c *CookbookService) Download(name, version string) error {
```

Comment:
@MarkGibbons I added two public functions:
1) `Download()` - downloads a cookbook to the **current directory** on disk
1) `DownloadAt()` - downloads a cookbook to the **specified local directory** on disk

These two functions give the users flexibility to use either or. Is there any concern to add this `Download()` function?

Finally, about the redundant directory creation, since it is happening in the `downloadCookbookItems` function, there, we are creating other directories like:
```
cookbook_name/recipes
cookbook_name/files
cookbook_name/templates
...
```

Let me know if you have suggestions that I could apply to this change.

Reply:
Nice